robotlib/src/CMakeLists.txt

```diff
@@ -1,10 +1,14 @@
 add_library(robotlib
-    placeholder.cpp
+    proximity_handler.cpp
 )
 
+target_link_libraries(robotlib
+    PRIVATE "io-interface-raspi")
+
 target_include_directories(robotlib 
     PUBLIC ../include 
     PRIVATE .
+    PRIVATE ../../io-interface/include/
```

Comment:
should not be needed because the `target_link_libraries` will include automatically the header marked as public